modules/farm_fd2/src/entrypoints/direct_seeding/direct_seeding.submission.e2e.cy.js

```diff
@@ -1,4 +1,41 @@
+import * as farmosUtil from '../../../../../library/farmosUtil/farmosUtil.js';
+
 describe('Direct Seeding: Submission tests', () => {
+  let cropMap = null;
+  let fieldMap = null;
+  let equipMap = null;
+  let bedMap = null;
+  let valuesMap = null;
+  let maps = new Map();
+
+  before(() => {
```

Comment:
The `before` should not `return` a value.  You have it returning the `Map` of `Promises` from the fetches of the maps that you use in the test.  Instead, I think you should have some code to fetch each map that looks similar to:

```JavaScript
farmosUtil.getFieldNameToAssetMap().then((map) => {
      fieldMap = map;
    }
```

Also, the `promiseMap` doesn't seem to really be used for anything other than assigning the values and so it seems overly complicated for that purpose as compared to the above.  If the `promiseMap` is eliminated then the test can just use the map variables directly.